pkg/network/http/ebpf_main.go

```diff
@@ -192,3 +193,19 @@ func (e *ebpfProgram) Close() error {
 	}
 	return err
 }
+
+func enableRuntimeCompilation(c *config.Config) bool {
+	if !c.EnableRuntimeCompiler {
+		return false
+	}
+
+	// The runtime-compiled version of HTTP monitoring requires Kernel 4.6
+	// because we use the `bpf_skb_load_bytes` helper.
+	kversion, err := kernel.HostVersion()
+	if err != nil {
+		log.Warn("could not determine the current kernel version. falling back to pre-compiled program.")
+		return false
+	}
+
+	return kversion >= kernel.VersionCode(4, 6, 0)
```

Comment:
It looks like the `bpf_skb_load_bytes` was added in 4.5, not 4.6: https://github.com/torvalds/linux/commit/05c74e5e53f6cb07502c3e6a820f33e2777b6605

Reply:
oh, good catch!